### DISCUSSION TITLE
Using NeoDash in NextJS project as Component

### DISCUSSION THREAD

**Discussion by Ahmad8864** [neo4j-labs/neodash#647] [Q&A]
Hello,

Would it be possible to use a published NeoDash dashboard as a component/page then populate the dashboard with information for each user? Or possibly embed NeoDash into the project?

**Comment by adriens** [comment]
Hi @Ahmad8864 , could you please explain a bit further your case (for example on a real life example) and all the benefits you see ?

**Reply by Ahmad8864** [reply]
Hello @adriens 

The Neo4J DB is going to store information about users in an organization and it will track interactions between users (messages, meetings, etc). I hope to use NeoDash in the website to provide a "stats" page for each user, detailing who they interact with most, displaying a graph of all the users they interact with, and whats their most used form of communication, etc. NeoDash is one of the best data visualization tools for Neo4J, especially with the breadth of options it has, so I was hoping to use it within NextJS/React.

Would it be possible to make some sort of component where the cypher queries are the same for each users, but the components display a specific user's information depending on who is logged in?

Thank you

**Comment by BennuFire** [comment]
Hello hello,

What you are describing sounds more like a RBAC configuration on your db.
Are you using Neo4j EE?

About NeoDash as a React component, we haven't created enough micro
components in order to be able to expose the app that way, and it's not a
priority at the moment.

**Reply by Ahmad8864** [reply]
Thanks for the response @BennuFire

I'm not using Neo4J EE at the moment, just Professional.

Do you know of any Neo4J tools that can produce a similar result to NeoDash but can be used within a react app?

**Comment by BennuFire** [comment]
If you don't use Rbac you can't rely on the db in order to apply visibility
rules. In that case, you may need to build your own app including those
rules on the middleware.
